app/controllers/concerns/authenticable.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Authenticable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :authenticate_user_using_x_auth_token
+    before_action :authenticate_user!
+  end
+
+  private
+
+    def authenticate_user_using_x_auth_token
+      user_email = request.headers["X-Auth-Email"]
+      auth_token = request.headers["X-Auth-Token"].presence
```

Comment:
`presence` can give `nil` value which might make `secure_compare` fail.
```suggestion
      auth_token = request.headers["X-Auth-Token"].to_s   
```